### PR TITLE
smr: Fix compile warnings, util/sockets: Handle fi_mr_attr compatibility

### PR DESCRIPTION
**Pull request by shefty**
Fixes compile warnings in smr provider

Updates util and sockets providers to handle fi_mr_attr backwards compatibility.  Specifically, do not access new HMEM related fields unless FI_HMEM capability has been requested by the app.  And do not access auth fields unless API version is 1.5 or greater.

### CODE REVIEW

File: prov/util/src/util_mr_map.c [hunk 1]
```diff
@@ -219,10 +219,39 @@ static struct fi_ops ofi_mr_fi_ops = {
...
+
+	if (FI_VERSION_GE(user_version, FI_VERSION(1, 5))) {
+		cur_abi_attr->auth_key_size = user_attr->auth_key_size;
+		cur_abi_attr->auth_key = user_attr->auth_key;
+	} else {
+		cur_abi_attr->auth_key_size = 0;
+		cur_abi_attr->auth_key = NULL;
+	}
+
+	if (caps & FI_HMEM) {
```
Comment (iziemba) [inline prov/util/src/util_mr_map.c:241]: FI_HMEM is not listed as a domain capability. May have to change to ```FI_VERSION_GE(user_version, FI_VERSION(1, 10))```.

Comment (shefty) [reply prov/util/src/util_mr_map.c:241]: The version doesn't work, based on the man pages.  Relying on just the version ends up being difficult on developers.  If there's a feature that 1.10 offers that they want, we want to make it easy for them to use it without also forcing code changes in places that they don't care about.  For example, if they update to 1.10 to get collective support, they shouldn't need to change how they register memory as a result.

FI_HMEM support isn't a domain capability, but it's an fi_info cap.  That's passed in when opening the domain, which gets captured in the util 'info_domain_caps'.

**Comment by shefty** [comment]
Intel CI was in accessible git repo.  AWS failure looks unrelated, but a real failure was detected.  Captured in github issue.

**Comment by shefty** [comment]
bot:aws:retest

**Pull request merged**